src/hooks/useHandleTransfer.tsx

```diff
@@ -308,207 +321,132 @@ async function evm(
+        signer
+      );
 
-          const receipt = await tx.wait();
+      const amountNormalizeAmount = deNormalizeAmount(
+        normalizeAmount(transferAmountParsed, decimals),
+        decimals
+      );
 
-          dispatch(
-            setTransferTx({
-              id: receipt.transactionHash,
-              block: receipt.blockNumber,
-            })
-          );
+      const estimateGas = await L2WormholeGateway.estimateGas.sendTbtc(
+        amountNormalizeAmount,
+        recipientChain,
+        recipientAddress,
+        THRESHOLD_ARBITER_FEE,
+        THRESHOLD_NONCE
+      );
 
-          enqueueSnackbar(null, {
-            content: <Alert severity="success">Transaction confirmed</Alert>,
-          });
+      // We increase the gas limit estimation here by a factor of 10% to account for
+      // some faulty public JSON-RPC endpoints.
+      const gasLimit = estimateGas.mul(1100).div(1000);
+      const overrides = {
+        gasLimit,
+        // We use the legacy tx envelope here to avoid triggering gas price autodetection using EIP1559 for polygon.
+        // EIP1559 is not actually implemented in polygon. The node is only API compatible but this breaks some clients
+        // like ethers when choosing fees automatically.
+        ...(chainId === CHAIN_ID_POLYGON && { type: 0 }),
+      };
 
-          const sequence = parseSequenceFromLogEth(
-            receipt,
-            getBridgeAddressForChain(chainId)
-          );
-          const emitterAddress = getEmitterAddressEth(
-            getTokenBridgeAddressForChain(chainId)
-          );
+      const tx = await L2WormholeGateway.sendTbtc(
+        amountNormalizeAmount,
+        recipientChain,
+        recipientAddress,
+        THRESHOLD_ARBITER_FEE,
+        THRESHOLD_NONCE,
+        overrides
+      );
 
-          await fetchSignedVAA(
-            chainId,
-            emitterAddress,
-            sequence,
-            enqueueSnackbar,
-            dispatch
-          );
-        } catch (e) {
-          console.error(parseError(e));
-        }
-      }
+      receipt = await tx.wait();
     } else {
-      const baseAmountParsed = parseUnits(amount, decimals);
-      const feeParsed = parseUnits(relayerFee || "0", decimals);
-      const transferAmountParsed = baseAmountParsed.add(feeParsed);
+      console.log("Regular flow! isTBTC?", isTBTC);
+
       // Klaytn requires specifying gasPrice
       const overrides =
         chainId === CHAIN_ID_KLAYTN
           ? { gasPrice: (await signer.getGasPrice()).toString() }
           : {};
-      const receipt = isNative
+
+      const additionalPayload = maybeAdditionalPayload();
+
+      receipt = isNative
         ? await transferFromEthNative(
             getTokenBridgeAddressForChain(chainId),
             signer,
             transferAmountParsed,
             recipientChain,
-            recipientAddress,
+            additionalPayload?.receivingContract || recipientAddress,
```

Comment:
Exactly, this is to access a different address than the default one, and also to add an additional payload if needed.

For now, it will be available only for TBTC flow, but shortly SEI stuff will be added to the same method as you can see on this [draft SEI PR](https://github.com/XLabs/portal-bridge-ui/pull/157) (I'll have to sync it with main branch after the merge of this one) 😄